_health-care/_js/_components/full_name.jsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import _ from 'lodash';
+
+class FullName extends React.Component {
+  constructor() {
+    super();
+    this.state = {hasError: false};
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  componentWillMount() {
+    this.id = _.uniqueId('date-input-');
+  }
+
+  handleChange() {
+    const name = {
+      first: this.refs.first.value,
+      middle: this.refs.middle.value,
+      last: this.refs.last.value,
+      suffix: this.refs.suffix.value
+    }
+
+    if (!this.validate(name.first, name.last)) {
+      this.setState({hasError: true});
+    } else {
+      this.setState({hasError: false});
+    }
+
+    this.props.onUserInput(name);
```

Comment:
Hmm...good point.  If you have a better pattern, feel free to fix all the components.

FYI, I want to enable https://facebook.github.io/react/docs/perf.html to see where time is going before jumping too far down the rabbit hole.  Optimizing w/o measurement (esp on a new framework) is kinda throwing darts blind.